libtransmission/rpc-server.h

```diff
@@ -43,7 +78,7 @@ class tr_rpc_server
     std::string whitelistStr;
     std::string url;
 
-    struct tr_address bindAddress;
+    struct tr_rpc_address bindAddress;
```

Comment:
I'm still not in love with all the above content being added to the header file just to make this line possible. Header files should contain as little as possible, because everything there is increased code coupling.

Optional, but could this be changed into a `struct tr_rpc_address*` whose lifecycle is managed in the .cc file just as httpd and start_retry_timer are?

Reply:
Might be misunderstanding you here, but you mean just leave a declaration in the header and move all of this to `rpc-server.cc`? That sounds like a much better idea than what I did.

Reply:
Yes, that's exactly what I mean! Sorry I was unclear. The basic principle being that we should share as little implementation detail as possible to the rest of the code.

I'd be OK with changing this to a pointer, but FWIW if you really want to go nuts with it you could use the [pimpl idiom](https://en.cppreference.com/w/cpp/language/pimpl) the same way that @mikedld did throughout the gtk/ client when he migrated it to gtkmm, and which I've started stealing, e.g. in tr_web.cc. That is not a requirement for this PR though; it would be a lot of scope creep.

Reply:
Gonna be honest with you that's really stretching my C++ knowledge, and allocation and keeping track of memory already stretches my C knowledge. In principle that sounds like the right way to go, for doing more advanced stuff and eventually ridding the RPC server of its evhttp dep, that seems required. For right now, however, I'd like to not break anything.